### PR TITLE
Add combo box control

### PR DESCRIPTION
**Pull request by peter-ha**
image

### CODE REVIEW

**Comment by peter-ha** [comment]
The ripple node role is not implemented for now, because this will only become more important once we have text input in the combo box.

File: src/controls/QskComboBox.cpp [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+QSK_SYSTEM_STATE( QskComboBox, Pressed, QskAspect::FirstSystemState << 1 )
+QSK_SYSTEM_STATE( QskComboBox, MenuOpen, QskAspect::FirstSystemState << 2 )
+
+class QskComboBox::PrivateData
+{
+  public:
+    PrivateData( QskComboBox* const box )
+        : menu( new QskMenu( box ) )
+    {
+        menu->setPopupFlag( QskPopup::DeleteOnClose, false );
```
Comment (uwerat) [inline src/controls/QskComboBox.cpp:29]: menu becomes a dangling pointer after the popup has been closed ( DeleteOnClose ). Guess there will be crashes, when using it a second time ?

File: src/controls/QskComboBox.cpp [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    else
+    {
+        hidePopup();
+    }
+
+    setSkinStateFlag( MenuOpen, on );
+    Q_EMIT menuOpenChanged( on );
+}
+
+bool QskComboBox::isMenuOpen() const
```
[the remark this answers is not in the document]
Comment (peter-ha) [reply src/controls/QskComboBox.cpp]: fixed with new patch set

File: src/controls/QskComboBox.cpp (hunk 1 above)
Comment (peter-ha) [reply src/controls/QskComboBox.cpp:29]: No, we are setting the flag to false, so it will still be around after closing, right?

**Pull request merged**